src/cdmtaskserviceclient/client.py

```diff
@@ -389,7 +398,14 @@ def wait_for_completion(
                     self._client._log.info(
                         f"Job {self.id} transitioned from {job_state} to {state}"
                     )
-                if state in ("complete", "error"):
+                if state == "error":
+                    return res
+                elif wait_for_event_importer:
+                    # if either of the event completion keys are in the admin_meta dict
+                    if _EVENT_COMPLETION_KEYS & res["admin_meta"].keys():
```

Comment:
I don't recall what does cts job status return - but does "admin_meta" always exist? Can `res["admin_meta"]` return None? 

You asked me to use the OpenAPI generator - why you’re not using it in your code and are opting for the simpler `CTSClient` approach instead?

Reply:
> does "admin_meta" always exist? 

Yes, pydantic defaults to {}

> You asked me to use the OpenAPI generator - why you’re not using it in your code 

Well, I suggested you try it. And IIUC we both seem to agree the results aren't great, so I didn't go down that route